### PR TITLE
WIP: RCHAIN-3136

### PR DESCRIPTION
**Pull request by JosDenmark**
## Overview
The last bit of `PoS.rho` that has to be done is generation of the unforgeable name. In the past, the `RegistrySigGen` has been used, but @odeac mentioned that this doesn't work. Instead, we deploy the contract with a print statement to determine the actual value of the first name generated, then use that in the registry insert. After that, tests have to be written for the contract. Hence, WIP.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3136

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -210,7 +210,7 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
                 deploy
               )
               _                                      <- runtime.shortLeashParams.setParams(codeHash, phloPrice, userId, timestamp)
-              injResult                              <- doInj(deploy, runtime.reducer, runtime.errorLog)(runtime.cost)
+              injResult                              <- doInj(deploy, runtime)
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: :ok_hand:

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala [hunk 2]
```diff
@@ -1910,4 +1911,136 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
...
+        } yield space.store.toMap
+        Await.result(inspectTask.runToFuture, 3.seconds)
+    }
+    result should be(mutable.HashMap.empty)
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(
+      Vector(MethodArgumentNumberMismatch("toList", 0, 1))
+    )
+  }
+
+  it should "transform Set(1, 2, 3) into a [1, 2, 3]" in {
```
Comment (ArturGajowy) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala]: All those tests should be expressed in rholang.

**Comment by ArturGajowy** [comment]
I'll make sure the PoS.rho contract is available in the registry in tests and genesis (production).

**Comment by ArturGajowy** [comment]
I've changed the identifier used a in PoS.rho to `PoS` (was MakePoS) to avoid confusion

**Comment by ArturGajowy** [comment]
> The last bit of PoS.rho that has to be done is generation of the unforgeable name. In the past, the RegistrySigGen has been used, but @odeac mentioned that this doesn't work.

I'm not sure what @odeac told you, but RegistrySigGen does work. It only needs the actual unforgable name to be passed - but it's enough to print it from a contract that's been deployed with known `sk, timestamp`, and pass all those to the tool.

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 3]
```diff
@@ -250,7 +250,7 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
             for {
               _         <- runtime.shortLeashParams.setParams(codeHash, phloPrice, userId, timestamp)
               _         <- runtime.replaySpace.rig(hash, log.toList)
-              injResult <- doInj(deploy, runtime.replayReducer, runtime.errorLog)(runtime.cost)
+              injResult <- doInj(deploy, runtime)
```
Comment (lukasz-golebiewski) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: This line used to forward the `replayReducer` into the `doInj` method. After your change the `doInj` defaults to using the `reducer`

Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: good catch! :+1:

**Pull request closed without merging**